mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -129,7 +131,9 @@ class KMeansModel private[ml] (
   @Since("2.0.0")
   def computeCost(dataset: Dataset[_]): Double = {
     SchemaUtils.checkColumnType(dataset.schema, $(featuresCol), new VectorUDT)
-    val data = dataset.select(col($(featuresCol))).rdd.map { case Row(point: Vector) => point }
+    val data = dataset.select(col($(featuresCol))).rdd.map { case Row(point: Vector) =>
+      OldVectors.fromML(point)
```

Comment:
@mengxr Implicit conversion doesn't work things like those. We still need manually convert them. But I agree that some of the code can be simplified by implicit which I will push in the next commit.